test/runner.js

```diff
@@ -12,46 +12,60 @@ const debug = require( 'debug' )( 'test-runner' ),
 	Mocha = require( 'mocha' ),
 	path = require( 'path' ),
 	program = require( 'commander' ),
+	watchFile = require( 'fs' ).watchFile,
 	chalk = require( 'chalk' );
 
-/**
- * Internal dependencies
- */
-const boot = require( './boot-test' ),
-	setup = require( './setup' );
-
 program
 	.usage( '[options] [files]' )
 	.option( '-R, --reporter <name>', 'specify the reporter to use', 'spec' )
 	.option( '-t, --node-total <n>', 'specify the node total to use', parseInt )
 	.option( '-i, --node-index <n>', 'specify the node index to use', parseInt )
-	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' );
+	.option( '-g, --grep <pattern>', 'only run tests matching <pattern>' )
+	.option( '-w, --watch', 'watch files for changes' );
 
 program.name = 'runner';
 
 program.parse( process.argv );
 
-const mocha = new Mocha( {
-	ui: 'bdd',
-	reporter: program.reporter
-} );
+const getMocha = function() {
+	/* These internal dependencies are not at the top of the file
+	 * because we need to re-require them every time we invalidate the cache.
+	 *
+	 * We invalidate the cache because Mocha has trouble running a test-suite
+	 * more than once:  https://github.com/mochajs/mocha/issues/995
+	 */
+	Object.keys( require.cache ).forEach( ( key ) => {
```

Comment:
In the linked Mocha examples they purge cache only for reruns. I must disagree that it makes our code more simple, because you can move all logic to a different place in a file (line 138 at the moment) and it will work exactly the same. My concern here is that purging the cache isn't the part of usual Mocha setup, but current implementation can lead to that conclusion for someone who doesn't know the context.